src/cargo/ops/cargo_add/mod.rs

```diff
@@ -378,7 +407,11 @@ fn resolve_dependency(
         };
         selected
     } else if let Some(raw_path) = &arg.path {
-        let path = paths::normalize_path(&std::env::current_dir()?.join(raw_path));
+        let mut relative_path = Path::new(raw_path);
+        if relative_path.ends_with("Cargo.toml") {
+            relative_path = relative_path.parent().unwrap();
```

Comment:
If you'd like to speed up this PR, I'd recommend removing this as it is a new feature that adds to our compatibility surface that mirrors other features,

Reply:
I've just changed it to an error. I feel that if I make it a separate PR, you'll ask to make it consistent with `install`, and install uses `SourceId` here, and that would require a whole refactoring of both commands, and that's too much for me.